tests/output/ConsoleInteractionTest.py

```diff
@@ -549,6 +553,43 @@ def test_ask_for_actions_and_apply5(self):
             self.assertEqual(generator.last_input, 1)
             self.assertNotIn('ChainPatchAction', failed_actions)
 
+    def test_ask_for_action_generate(self):
+        failed_actions = set()
+        action = ApplyPatchAction()
+        action2 = GeneratePatchesAction()
+        args = [self.console_printer, Section(''),
+                [action.get_metadata(), action2.get_metadata()],
+                {'ApplyPatchAction': action, 'GeneratePatchesAction': action2},
+                failed_actions, Result('origin', 'message'), {}, {}, {}]
+
+        with simulate_console_inputs('g', 5) as generator:
+            ask_for_action_and_apply(*args)
+            self.assertEqual(generator.last_input, 0)
+            self.assertNotIn('GeneratePatchAction', failed_actions)
+
+    def test_ask_for_action_generate2(self):
```

Comment:
Can you please give some meaningful name to test case instead of just suffixing 2 to avoid confusion ? (or may be suffix something meaningful instead of just number)